server/src/routes/api/v2/contacts.js

```diff
@@ -0,0 +1,72 @@
+const router = require('express').Router()
+const mongoose = require('mongoose')
+const { ObjectId } = mongoose.Types
+const auth = require('@routes/auth')
+const UserWallet = mongoose.model('UserWallet')
+const Contact = mongoose.model('Contact')
+const { concat, compact, indexOf } = require('lodash')
+
+/**
+ * @api {post} /contacts/ Sync contacts list
+ * @apiName SyncContacts
+ * @apiGroup Contacts
+ * @apiDescription Sync contacts list
+ *
+ * @apiHeader {String} Authorization JWT Authorization in a format "Bearer {jwtToken}"
+ *
+ * @apiParam {String[]} contacts phone numbers list
+ *
+ * @apiSuccess {Object[]} new contacts list (phoneNumber, account)
+ * @apiSuccess {Number} nonce
+ */
+router.post('/', auth.required, async (req, res) => {
+  const { phoneNumber, accountAddress } = req.user
+  const { contacts } = req.body
+  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress }).populate('contacts')
+  const userWalletContactIds = userWallet.contacts.map(obj => obj._id)
+  const userWalletContactPhoneNumbers = userWallet.contacts.map(obj => obj.phoneNumber)
```

Comment:
Make sense to do parts of the loop in one query if possible

Reply:
Can you explain>